rustworkx-core/src/centrality.rs

```diff
@@ -1095,42 +1095,50 @@ mod test_katz_centrality {
 /// let dg = petgraph::graph::DiGraph::<i32, ()>::from_edges(&[
 ///     (0, 4), (1, 2), (2, 3), (3, 4), (1, 4)
 /// ]);
-/// let output = closeness_centrality(&dg, true);
+/// let output = closeness_centrality(&dg, true, 200);
 /// assert_eq!(
 ///     vec![Some(0.), Some(0.), Some(1./4.), Some(1./3.), Some(4./5.)],
 ///     output
 /// );
 /// ```
-pub fn closeness_centrality<G>(graph: G, wf_improved: bool) -> Vec<Option<f64>>
+pub fn closeness_centrality<G>(
+    graph: G,
+    wf_improved: bool,
+    parallel_threshold: usize,
+) -> Vec<Option<f64>>
 where
     G: NodeIndexable
         + IntoNodeIdentifiers
         + GraphBase
         + IntoEdges
         + Visitable
         + NodeCount
-        + IntoEdgesDirected,
-    G::NodeId: std::hash::Hash + Eq,
+        + IntoEdgesDirected
+        + std::marker::Sync,
+    G::NodeId: Eq + Hash + Send,
+    G::EdgeId: Eq + Hash + Send,
 {
     let max_index = graph.node_bound();
-    let mut closeness: Vec<Option<f64>> = vec![None; max_index];
-    for node_s in graph.node_identifiers() {
+    let node_indices: Vec<G::NodeId> = graph.node_identifiers().collect();
+    let closeness: Mutex<Vec<Option<f64>>> = Mutex::new(vec![None; max_index]);
```

Comment:
Instead of doing this as an vec wrapped in a mutex could you do this as a collection over the parallel iterator? It looks like each iteration is the node index so I think you should be able to do this with `collect()` and not need the mutex.

Reply:
I will try, I don't remeber explicitly if `collect()` on a parallel iterator from `par_iter()` preserves order. So I will need to check. That is one of the only advantages of the the mutex approach.

An `enumerate()` could also work here but then I will need to fight with Rust's type system I think

Reply:
`par_iter()` does preserve order for `collect()`, so I don't think it should be too difficult.

Reply:
Indeed: https://github.com/rayon-rs/rayon/issues/669

It is only `par_bridge` that doesn't preserve the order

Reply:
Done.